browser/ui/views/speedreader/speedreader_icon_view.cc

```diff
@@ -92,32 +76,44 @@ void SpeedreaderIconView::UpdateImpl() {
             GetOmniboxColor(theme_provider, OmniboxPart::RESULTS_ICON);
         SetIconColor(icon_color_default);
       }
+      UpdateIconImage();
     }
   }
 
   if (is_distilled) {
-    const int label_id = state == DistillState::kReaderMode
-                             ? IDS_ICON_READER_MODE_LABEL
-                             : IDS_ICON_SPEEDREADER_MODE_LABEL;
-    SetLabel(l10n_util::GetStringUTF16(label_id));
-    UpdateLabelColors();
+    UpdateIconImage();
     if (theme_provider)
       SetIconColor(theme_provider->GetColor(
           BraveThemeProperties::COLOR_SPEEDREADER_ICON));
     SetVisible(true);
-    label()->SetVisible(true);
   }
-
-  Observe(contents);
 }
 
 const gfx::VectorIcon& SpeedreaderIconView::GetVectorIcon() const {
-  return kSpeedreaderIcon;
+  const DistillState state = GetDistillState();
+  if (state == DistillState::kSpeedreaderMode ||
+      state == DistillState::kSpeedreaderOnDisabledPage) {
+    return kBraveSpeedreaderModeIcon;
+  } else {
+    return kBraveReaderModeIcon;
+  }
 }
 
 std::u16string SpeedreaderIconView::GetTextForTooltipAndAccessibleName() const {
-  return l10n_util::GetStringUTF16(GetActive() ? IDS_EXIT_DISTILLED_PAGE
-                                               : IDS_DISTILL_PAGE);
+  int id;
+  const DistillState state = GetDistillState();
+  switch (state) {
+    case DistillState::kSpeedreaderMode:
+    case DistillState::kSpeedreaderOnDisabledPage:
+      id = IDS_SPEEDREADER_ICON_SPEEDREADER_SETTINGS;
+      break;
+    case DistillState::kReaderMode:
+      id = IDS_SPEEDREADER_ICON_TURN_OFF_READER_MODE;
```

Comment:
note that this state is assuming the behavior of https://github.com/brave/brave-core/pull/9393 being merged

Reply:
merged 9393 and rebased before merge. resolving.